src/platform/Darwin/PlatformManagerImpl.mm

```diff
@@ -178,5 +180,45 @@
 #endif // CONFIG_NETWORK_LAYER_BLE
     }
 
+    bool PlatformManagerImpl::RegisterSignalHandler(int sig, dispatch_block_t block)
+    {
+        VerifyOrReturnValue(CanCastTo<uintptr_t>(sig), false);
+        VerifyOrReturnValue(mSignalSources.find(sig) == mSignalSources.end(), false); // Already registered
+
+        signal(sig, SIG_IGN); // Ignore signal so GCD can catch it.
+
+        __auto_type source = dispatch_source_create(DISPATCH_SOURCE_TYPE_SIGNAL, static_cast<uintptr_t>(sig), 0, mSignalQueue);
+        dispatch_source_set_event_handler(source, ^{
+            dispatch_async(mWorkQueue, block);
+        });
+
+        mSignalSources[sig] = source;
+        dispatch_resume(source);
+
+        return true;
+    }
+
+    bool PlatformManagerImpl::UnregisterSignalHandler(int sig)
+    {
+        VerifyOrReturnValue(CanCastTo<uintptr_t>(sig), false);
+        VerifyOrReturnValue(mSignalSources.find(sig) != mSignalSources.end(), false); // Not registered
+
+        __auto_type it = mSignalSources.find(sig);
```

Comment:
Might be clearer to do this, then `VerifyOrReturnValue(it != mSignalSources.end(), false); // Not registered` ?